base/server/src/main/java/com/netscape/cms/authorization/DirAclAuthz.java

```diff
@@ -179,15 +189,74 @@ public void loadACLs() throws EBaseException {
             while (en.hasMoreElements()) {
                 String acl = en.nextElement();
                 logger.info("DirAclAuthz: - " + acl);
-                addACLs(acl);
+
+                // call super.addACLs() directly to avoid re-entering loadACLs()
+                super.addACLs(acl);
             }
 
         } catch (LDAPException e) {
             throw new EACLsException(CMS.getUserMessage("CMS_ACL_CONNECT_LDAP_FAIL", e.getMessage()), e);
 
+        } catch (ELdapException e) {
+            throw new EACLsException(CMS.getUserMessage("CMS_ACL_CONNECT_LDAP_FAIL", e.getMessage()), e);
+
         } finally {
             returnConn(conn);
         }
+
+        loaded = true;
+    }
+
+    @Override
+    public void addACLs(String resACLs) throws EACLsException {
+        loadACLs();
```

Comment:
`loadACLs()`is a syncronized method so this require some overhead to be called. Since the first operation is to check a boolean, for performance it would be better to check the boolean also here and call the method only if `loaded` is `False`. The same apply to the following methods.

Reply:
Good point. I updated the PR.